modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -628,10 +628,41 @@ private void _validateParameterObjectFieldId(
 					" requires parameter");
 		}
 
-		if ((parameterObjectFieldId != null) && !parameterRequired) {
-			throw new ObjectRelationshipParameterObjectFieldIdException(
-				"Object definition " + objectDefinition.getName() +
-					" does not allow parameter");
+		if (parameterObjectFieldId != null) {
+			if (!parameterRequired) {
+				throw new ObjectRelationshipParameterObjectFieldIdException(
+					"Object definition " + objectDefinition.getName() +
+						" does not allow parameter");
+			}
+
+			ObjectField objectField = _objectFieldLocalService.getObjectField(
+				parameterObjectFieldId);
+
+			if (objectField == null) {
+				throw new ObjectRelationshipParameterObjectFieldIdException(
+					"Parameter object field " + parameterObjectFieldId +
+						" does not exist");
+			}
+
+			if (objectDefinition.getObjectDefinitionId() !=
+					objectField.getObjectDefinitionId()) {
+
+				throw new ObjectRelationshipParameterObjectFieldIdException(
+					StringBundler.concat(
+						"Parameter object field ", parameterObjectFieldId,
+						" does not belong to object definition ",
+						objectDefinition.getName()));
+			}
+
+			if (Objects.equals(
+					objectField.getBusinessType(),
+					ObjectFieldConstants.BUSINESS_TYPE_RELATIONSHIP)) {
+
```

Comment:
Shouldn't it be `!Objects.equals(..)` instead?

Reply:
@carolmariaabb, I moved this change to LPS-156125. If you want to check, you can see here in this same pull request, but I already sent it along with the other PR that contains Brian's requests.